opwen_email_client/webapp/actions.py

```diff
@@ -223,3 +229,66 @@ def __call__(self):
         finally:
             if connection is not None:
                 connection.terminate()
+
+
+class ClientRegister(object):
+    def __init__(self, client_name: str, username: str, access_token: str, logger: Logger):
+        self._client_name = client_name
+        self._github_username = username
+        self._github_access_token = access_token
+        self._log = logger or getLogger(__name__)
+
+    @property
+    def client_domain(self):
+        return '{}.{}'.format(self._client_name, 'lokole.ca')
+
+    @property
+    def client_url_create(self):
+        return 'https://{}/api/email/register/'.format('mailserver.lokole.ca')
+
+    @property
+    def client_url_details(self):
+        return 'https://{}/api/email/register/{}'.format('mailserver.lokole.ca', self.client_domain)
+
+    def __call__(self):
+        create_request_payload = dumps({'domain': self.client_domain}).encode('utf-8')
+        create_request = Request(self.client_url_create)
+        create_request.add_header('Content-Type', 'application/json; charset=utf-8')
+        create_request.add_header('Content-Length', str(len(create_request_payload)))
+        create_request.add_header('Authorization', 'Bearer {}'.format(self._github_access_token))
+
+        try:
+            with urlopen(create_request, create_request_payload):  # nosec
+                pass
+        except HTTPError as ex:
+            self._log.exception('Unable to register client {client_name}: [{status_code}] {message}'.format(
+                client_name=self._client_name,
+                status_code=ex.code,
+                message=ex.read().decode('utf-8').strip()
+            ))
```

Comment:
I wonder if this block of code should run synchonously inside of the RegisterForm so that we can indicate an error on the form to the user if their credentials are incorrect, i.e. instead of logging the exception to actually show it to the user.

Reply:
I see, yes that would be the best course of action